src/sync/installation.ts

```diff
@@ -351,13 +351,17 @@ export const handleBackfillError = async (
 	}
 
 	if (String(err).includes("connect ETIMEDOUT")) {
-		// There was a network connection issue.
-		// Add the job back to the queue with a 5 second delay
 		logger.warn("ETIMEDOUT error, retrying in 5 seconds");
 		scheduleNextTask(5_000);
 		return;
 	}
 
+	if (String(err).includes("connect ECONNREFUSED")) {
```

Comment:
This could be too broad, as GHE customer might also have this ECONNREFUSED if their firewall decide the block our connections.

Can we do a try/wrap in the `EncryptionClient` file to throw a more specific error, like `err.type === 'Encyprtion'`?

Reply:
TBH I don't want to leak knowledge all around the place.

If you check a few lines above, there's already "timeout" clause – you could argue that could be due to firewall rules, too. If we want to make it better, we'd need to change it as well, but I doubt that should be done on this level. This is a global error handler: could be a Jira error, a github error, a cryptor error, just any network error.

Please ping me in slack if you have a strong opinion or would like to discuss in more details :)